python3/uproctrace/gui.py

```diff
@@ -352,6 +354,51 @@ def onTreeToggled(self, _widget):
         # re-populate processes view
         self.populateProcesses()
 
+    def onNotificationClose(self, _widget):
+        """
+        Notification close button pressed: close the notification
+        """
+        self.closeNotification()
+
+    def closeNotification(self):
+        """
+        Closes the notification and kills the timeout if set
+        """
+        self.notifier.set_reveal_child(False)
+        if self.notifier_timeout is not None:
+            GObject.source_remove(self.notifier_timeout)
+            self.notifier_timeout = None
+
+    def showNotification(self, message: str, timeout_ms: int = None):
+        """
+        Shows a notification with the given message text, if timeout (in ms)
+        is provided also sets a timeout to close the notification
+        """
+        self.closeNotification()
+
+        self.notifier_msg.set_text(message)
+        self.notifier.set_reveal_child(True)
+
+        if timeout_ms is not None:
+            self.notifier_timeout = GObject.timeout_add(
+                timeout_ms, self.closeNotification)
+
+    def storeInClipboard(self, string: str):
+        """
+        Stores a string in the clipboard
+        """
+        self.clipboard.set_text(string, -1)
+        self.clipboard.store()
+
+    def storeInClipboardAndNotify(self, string: str):
+        """
+        Stores a string in the clipboard and shows a "copied to clipboard"
+        notification
+        """
+        self.storeInClipboard(string)
+        msg = string if len(string) <= 100 else string[:97] + "..."
+        self.showNotification(f"'{msg:s}'\nCopied to clipboard", 1000)
```

Comment:
`f"{repr(msg):s}\n...` would even escape backslashs and single quotes in the string.